ogr/ogrsf_frmts/gml/parsexsd.cpp

```diff
@@ -957,7 +958,9 @@ static void CPLXMLSchemaResolveInclude(const char *pszMainSchemaLocation,
         for (; psThis != nullptr; psThis = psThis->psNext)
         {
             if (psThis->eType == CXT_Element &&
-                EQUAL(psThis->pszValue, "include"))
+                (EQUAL(psThis->pszValue, "include") ||
+                 (useSchemaImports == TRUE &&
+                  EQUAL(psThis->pszValue, "import"))))
```

Comment:
Could you add below an exemption for base GML referenced schemas that won't bring anything when importing because they don't contain any feature types. So schemaLocations containing:
- /gml/3.2.1/
- gml/3.1.1/
- /gml/2.1.2/
- /gmlsfProfile/

For things like:
```
autotest/ogr/data/gml/expected_gml_gml32.xsd:<xs:import namespace="http://www.opengis.net/gml/3.2" schemaLocation="http://schemas.opengis.net/gml/3.2.1/gml.xsd"/>
autotest/ogr/data/gml/expected_gml_gml32.xsd:<xs:import namespace="http://www.opengis.net/gmlsf/2.0" schemaLocation="http://schemas.opengis.net/gmlsfProfile/2.0/gmlsfLevels.xsd"/>
autotest/ogr/data/gml/testLookForSimpleType.xsd:<xs:import namespace="http://www.opengis.net/gml" schemaLocation="http://schemas.opengis.net/gml/2.1.2/feature.xsd"/>
```

Reply:
I have provided a fix to exclude gml imports but have not provided any unit test for the exclusion of GML as I wasn't sure how that can be tested. Please let me know your feedback.